pkg/operator2/idp.go

```diff
@@ -216,33 +216,37 @@ func (c *authOperator) convertProviderConfigToIDPData(providerConfig *configv1.I
 	return data, nil
 }
 
-func (c *authOperator) discoverOpenIDURLs(issuer, key string, ca configv1.ConfigMapNameReference) (*osinv1.OpenIDURLs, error) {
+// discoverOpenIDConfig retrieves basic information about an OIDC server with hostname
+// given by the `issuer` argument. It returns the authentication, token and user
+// info urls in *osinv1.OpenIDUrls structure, and a boolean value telling whether
+// our challenge flows can be used against such a server.
+func (c *authOperator) discoverOpenIDConfig(issuer, key string, ca configv1.ConfigMapNameReference) (*osinv1.OpenIDURLs, bool, error) {
```

Comment:
nit: Maybe simplify both this change and future maintenance by returning a module-local struct with fields for url and whether password grants are permitted? That would also have the advantage of being self documenting vs requiring as much detail in the docstring.